sdk/keyvault/keyvault-certificates/test/CRUD.test.ts

```diff
@@ -272,37 +272,68 @@ describe("Certificates client - create, read, update and delete", () => {
     );
   });
 
-  it("can get a deleted certificate", async function() {
-    const certificateName = testClient.formatName(`${prefix}-${this!.test!.title}-${suffix}`);
-    await client.beginCreateCertificate(
-      certificateName,
-      basicCertificatePolicy,
-      testPollerProperties
-    );
-    const deletePoller = await client.beginDeleteCertificate(certificateName, testPollerProperties);
-    const deletedCertificate = await deletePoller.pollUntilDone();
-    assert.equal(
-      deletedCertificate.properties.name,
-      certificateName,
-      "Unexpected certificate name in result from getCertificate()."
-    );
-    await testClient.purgeCertificate(certificateName);
-  });
+  describe("can get a deleted certificate", () => {
+    it("using beginDeleteCertificate's poller", async function() {
+      const certificateName = testClient.formatName(`${prefix}-${this!.test!.title}-${suffix}`);
+      await client.beginCreateCertificate(
+        certificateName,
+        basicCertificatePolicy,
+        testPollerProperties
+      );
 
-  it("can get a deleted certificate (Non Existing)", async function() {
-    const certificateName = testClient.formatName(`${prefix}-${this!.test!.title}-${suffix}`);
-    let error;
-    try {
-      await client.beginDeleteCertificate(certificateName, testPollerProperties);
-      throw Error("Expecting an error but not catching one.");
-    } catch (e) {
-      error = e;
-    }
-    assert.equal(
-      error.message,
-      `Certificate not found: ${certificateName}`,
-      "Unexpected certificate name in result from getKey()."
-    );
+      const deletePoller = await client.beginDeleteCertificate(
+        certificateName,
+        testPollerProperties
+      );
+      let deletedCertificate = await deletePoller.pollUntilDone();
+      assert.equal(
+        deletedCertificate.name,
+        certificateName,
+        "Unexpected certificate name in result from getCertificate()."
+      );
+
+      await testClient.purgeCertificate(certificateName);
+    });
+
+    it("using getDeletedCertificate", async function() {
+      const certificateName = testClient.formatName(`${prefix}-${this!.test!.title}-${suffix}`);
+      await client.beginCreateCertificate(
+        certificateName,
+        basicCertificatePolicy,
+        testPollerProperties
+      );
+
+      const deletePoller = await client.beginDeleteCertificate(
+        certificateName,
+        testPollerProperties
+      );
+      await deletePoller.pollUntilDone();
+
+      const deletedCertificate = await client.getDeletedCertificate(certificateName);
+      assert.equal(
+        deletedCertificate.name,
+        certificateName,
+        "Unexpected certificate name in result from getCertificate()."
+      );
+
+      await testClient.purgeCertificate(certificateName);
+    });
+
+    it("can get a deleted certificate (Non Existing)", async function() {
```

Comment:
So keyvaults treats deleting a key that never existed differently than a key that used to exist?

Reply:
Yes, but in the sense that we should throw appropriately if a user wants to retrieve a deleted key that doesn't exist.

Reply:
> we should throw appropriately if a user wants to retrieve a deleted key that doesn't exist
This error is coming from the service and not the SDK right? If a key is deleted and a year later I try to delete it, would I get a certificate not found error?

Reply:
this error comes indeed from the service, not from the SDK. The test is just meant as an example of what happens if this fails. We might not do this in the future! I'm working on defining how to improve our tests throughout this repository as an effort independent of this PR.

Reply:
I'll be able to show something regarding the test improvements by next week's TS meeting. 🌞

Reply:
Should the test be renamed to something like:
```suggestion
    it("can not get a certificate that never existed", async function() {
```